internal/server/rest.go

```diff
@@ -283,17 +283,18 @@ func rulesHandler(w http.ResponseWriter, r *http.Request) {
 		} else {
 			result = fmt.Sprintf("Rule %s was created successfully.", r.Id)
 		}
-		//Start the rule
-		rs, err := createRuleState(r)
-		if err != nil {
-			result = err.Error()
-		} else {
-			err = doStartRule(rs)
+		go func() {
+			//Start the rule
+			rs, err := createRuleState(r)
 			if err != nil {
 				result = err.Error()
+			} else {
+				err = doStartRule(rs)
+				if err != nil {
+					result = err.Error()
+				}
 			}
-		}
-
+		}()
 		w.WriteHeader(http.StatusCreated)
 		w.Write([]byte(result))
```

Comment:
now move these two function in go routine, if createRuleState(r) or doStartRule(rs) has error, result can not get the error message

Reply:
Logically the rule meta has created even when the rule cannot start correctly, so it is correct to return created. The start rule error is a runtime error which can be found by getStatus or log.